src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Strings.vb

```diff
@@ -3,48 +3,280 @@
+            '-------------------------------------------------------------
+            '  Notes:
+            '   VB6 order of execution
+            '   verify Start > 0 ==> vbIllegalFuncCall
+            '   verify lLen > 0 ==> vbIllegalFuncCall
+            '   return computed string
+            '-------------------------------------------------------------
+            If Start <= 0 Then
+                Throw New ArgumentException(SR.Argument_GTZero1, NameOf(Start))
+            ElseIf Length < 0 Then
+                Throw New ArgumentException(SR.Argument_GEZero1, NameOf(Length))
+            ElseIf Length = 0 OrElse [str] Is Nothing Then
+                Return ""
+            End If
+
+            Dim lStrLen As Integer
+
+            lStrLen = [str].Length
+
+            If Start > lStrLen Then
+                Return ""
+            ElseIf (Start + Length) > lStrLen Then
+                Return [str].Substring(Start - 1)
+            Else
+                Return [str].Substring(Start - 1, Length)
+            End If
+        End Function
+
+        Public Function Right(ByVal [str] As String, ByVal Length As Integer) As String
+            If Length < 0 Then
+                Throw New ArgumentException(SR.Argument_GEZero1, NameOf(Length))
+            End If
+
+            If Length = 0 OrElse [str] Is Nothing Then
+                Return ""
+            End If
+
+            Dim lStrLen As Integer
+
+            lStrLen = [str].Length
+
+            If Length >= lStrLen Then
+                Return [str]
+            End If
+
+            Return [str].Substring(lStrLen - Length, Length)
+        End Function
+
+        Public Function RTrim(ByVal [str] As String) As String
+            Try
+                If [str] Is Nothing OrElse str.Length = 0 Then
+                    Return ""
+                End If
+
+                Dim ch As Char
+                ch = str.Chars(str.Length - 1)
+
+                If ch = chSpace OrElse ch = chIntlSpace Then
+                    Return [str].TrimEnd(m_achIntlSpace)
+                End If
+                Return str
+            Catch ex As Exception
+                Throw ex
+            End Try
+        End Function
+
+        Public Function Trim(ByVal str As String) As String
+            Try
+                If str Is Nothing OrElse str.Length = 0 Then
+                    Return ""
+                End If
+
+                Dim ch As Char = str.Chars(0)
+                If ch = chSpace OrElse ch = chIntlSpace Then
+                    Return str.Trim(m_achIntlSpace)
+                Else
+                    ch = str.Chars(str.Length - 1)
+                    If ch = chSpace OrElse ch = chIntlSpace Then
+                        Return str.Trim(m_achIntlSpace)
+                    End If
+                End If
+                Return str
+
+            Catch ex As Exception
+                Throw ex
+            End Try
+        End Function
```

Comment:
I assume all of this is just copy and paste out of the original code?

Reply:
Yes